OpenRA.Mods.Cnc/SpriteLoaders/ShpTDLoader.cs

```diff
@@ -75,6 +75,95 @@ public class ShpTDSprite
 	{
 		enum Format : ushort { XORPrev = 0x20, XORLCW = 0x40, LCW = 0x80 }
 
+		class TrimmedFrame : ISpriteFrame
+		{
+			public SpriteFrameType Type => SpriteFrameType.Indexed8;
+			public Size Size { get; }
+			public Size FrameSize { get; }
+			public float2 Offset { get; }
+			public byte[] Data { get; }
+			public bool DisableExportPadding { get { return false; } }
+
+			public TrimmedFrame(ImageHeader header)
+			{
+				var origData = header.Data;
+				var origSize = header.Size;
+				var top = origSize.Height - 1;
+				var bottom = 0;
+				var left = origSize.Width - 1;
+				var right = 0;
+
+				// Scan frame data to find left-, top-, right-, bottom-most
+				// rows/columns with non-zero pixel data.
+				var i = 0;
+				for (var y = 0; y < origSize.Height; y++)
+				{
+					for (var x = 0; x < origSize.Width; x++, i++)
+					{
+						if (origData[i] != 0)
+						{
+							top = Math.Min(y, top);
+							bottom = Math.Max(y, bottom);
+							left = Math.Min(x, left);
+							right = Math.Max(x, right);
+						}
+					}
+				}
+
+				// Keep a 1px empty border to work avoid rounding issues in the gpu shader.
```

Comment:
ah, a relic of the past. It seems that from the lessons learned it was built into the sheetbuilder instead in #17581

Reply:
no, this line is necessary. Otherwise it cuts of hats like in #15350

<img width="90" alt="Screenshot 2024-03-12 at 19 53 15" src="https://github.com/OpenRA/OpenRA/assets/37534529/0fa7feb6-1e49-4f98-b90f-5cd12a377a32">